### DISCUSSION TITLE
1.32.1

### DISCUSSION THREAD

**Discussion by BlackDex** [dani-garcia/vaultwarden#5036] [General]
## Notable changes
- Fixed syncing/login with native mobile clients
- Added CLI option to backup SQLite database
- Email Template changes regarding invites, 2FA Incomplete logins, and new logins

## What's Changed
* Update GitHub Action Workflows by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4849
* Fix Duo Redirect not using path by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4862
* Fix manager in web-vault v2024.6.2 for collections by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4860
* Update email footer padding values by @dfunkt in https://github.com/dani-garcia/vaultwarden/pull/4838
* Remove unecessary email normalization by @Timshel in https://github.com/dani-garcia/vaultwarden/pull/4840
* Fix Vaultwarden Admin page error messages by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4869
* Update issue template by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4876
* remove overzealous sanity check by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4879
* Fix Login with device by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4878
* Switch to Whitelisting in .dockerignore by @Timshel in https://github.com/dani-garcia/vaultwarden/pull/4856
* Remove `version` from server config info by @zacknewman in https://github.com/dani-garcia/vaultwarden/pull/4885
* Update issue template by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4882
* Update crates (GHSA-wq9x-qwcq-mmgf) by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4889
* Updated security readme by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4892
* Allow custom umask setting by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4896
* Allow Org Master-Pw policy enforcement by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4899
* Allow enforcing Single Org with pw reset policy by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4903
* Add a CLI feature to backup the SQLite DB by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4906
* Update web-vault, crates and gha by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4909
* Add orgUserHasExistingUser parameters to org invite by @Timshel in https://github.com/dani-garcia/vaultwarden/pull/4827
* Update Rust version & crates by @dfunkt in https://github.com/dani-garcia/vaultwarden/pull/4928
* Fix sync with new native clients by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4932
* Fix collection update from native client by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4937
* fix invitation link via /admin by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4950
* Fix Pw History null dates by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4966
* fix 2fa policy check on registration by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4956
* Actually use Device Type for mails by @dfunkt in https://github.com/dani-garcia/vaultwarden/pull/4916
* remove backtics from postgresql migrations by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4968
* Fix Device Type column for 2FA migration by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4971
* Fix encrypted lastUsedDate by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4972
* Fix keyword collision in Rust 2024 and add new api/config value by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4975
* Add extra linting by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4977

## New Contributors
* @zacknewman made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4885

**Full Changelog**: https://github.com/dani-garcia/vaultwarden/compare/1.32.0...1.32.1

<hr /><em>This discussion was created from the release <a href='https://github.com/dani-garcia/vaultwarden/releases/tag/1.32.1'>1.32.1</a>.</em>

**Comment by dfunkt** [comment]
I'd say the Device Type template change would merit a mention in the notable changes section.
People should know that they need to update the `incomplete_2fa_login` and `new_device_logged_in` templates if they're using custom ones.

**Reply by BlackDex** [reply]
I was going to do that, but then somehow i didn't. I added it right now :).

**Comment by BJReplay** [comment]
Nice to see #4972 hit release so quickly!  Thanks for working with me to fix!

**Comment by xingpingcn** [comment]
> Added CLI option to backup SQLite database

how to use it?

**Reply by BlackDex** [reply]
```bash
/vaultwarden backup
# Or
docker exec -it vaultwarden /vaultwarden backup
# Or
kill -s USR1 pid-of-vaultwarden
```

I'm going to write a more detailed wiki soon.

**Comment by serrq** [comment]
Is it an Android client?

**Reply by BlackDex** [reply]
Is what an Android client?

**Comment by swe-lockdrives** [comment]
remove overzealous sanity check by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4879 

->

So the error that the organization ID is empty can easily occur again, right?

**Reply by stefan0xC** [reply]
what do you mean by easily?

**Reply by swe-lockdrives** [reply]
The sanity check was there to ensure that the problem could no longer occur -> https://github.com/dani-garcia/vaultwarden/discussions/5011


Now the sanity check has been removed in this release?

Now the error can happen again :(

**Reply by stefan0xC** [reply]
the sanity check was faulty because it did not take into account that when cloning an entry from an organization to an personal vault the client would send collection ids but not an `organization_id`. if you can think of a way to differentiate between the two cases where a missing organization id is legitimate and when it happens because the optional field has failed to be deserialized, please tell.
